src/OcspResponse.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/*
+ * Copyright (c) 2020-2021 Estonian Information System Authority
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+declare(strict_types=1);
+
+namespace web_eid\ocsp_php;
+
+use phpseclib3\File\ASN1;
+use UnexpectedValueException;
+use web_eid\ocsp_php\exceptions\OcspResponseDecodeException;
+use web_eid\ocsp_php\exceptions\OcspVerifyFailedException;
+use web_eid\ocsp_php\maps\OcspBasicResponseMap;
+use web_eid\ocsp_php\maps\OcspResponseMap;
+
+class OcspResponse
+{
+    private array $ocspResponse = [];
+    private string $revokeReason = "";
+
+    public function __construct(string $encodedBER)
+    {
+        $decoded = ASN1::decodeBER($encodedBER);
+        if (!$decoded[0]) {
+            throw new OcspResponseDecodeException();
+        }
+
+        $this->ocspResponse = ASN1::asn1map($decoded[0], OcspResponseMap::MAP, [
+            "response" => function ($encoded) {
+                return ASN1::asn1map(
+                    ASN1::decodeBER($encoded)[0],
+                    OcspBasicResponseMap::MAP
+                );
+            },
+        ]);
+    }
+
+    public function getResponse(): array
+    {
+        return $this->ocspResponse;
+    }
+
+    public function getBasicResponse(): OcspBasicResponse
+    {
+        if (
+            Ocsp::ID_PKIX_OCSP_BASIC_STRING !=
+            $this->ocspResponse["responseBytes"]["responseType"]
+        ) {
+            throw new UnexpectedValueException(
+                'responseType is not "id-pkix-ocsp-basic" but is ' .
+                    $this->ocspResponse["responseBytes"]["responseType"]
+            );
+        }
+
+        if (!$this->ocspResponse["responseBytes"]["response"]) {
+            throw new UnexpectedValueException(
+                "Could not decode OcspResponse->responseBytes->responseType"
```

Comment:
There's a mistake in the error message:
```suggestion
                "Could not decode OcspResponse->responseBytes->response"
```